platform/ios/src/MGLMapboxEvents.m

```diff
@@ -249,38 +257,61 @@ - (NSString *)instanceID {
 }
 
 - (void)userDefaultsDidChange:(NSNotification *)notification {
-    dispatch_async(dispatch_get_main_queue(), ^{
-        [self pauseOrResumeMetricsCollectionIfRequired];
-    });
+    
+    // Guard against over calling pause / resume if the values this implementation actually
+    // cares about have not changed
+    
+    if ([[notification object] isMemberOfClass:[NSUserDefaults class]]) {
```

Comment:
How about calling `-respondsToSelector:@selector(objectForKey:)` instead? I wasn't sure offhand whether this passed the `NSUserDefaults` instance or an `NSDictionary`, but this check would work in either case

Reply:
@akitchen I like the idea but was thinking about comparison speed when I wrote this since these changes are only really valuable when there are many calls. 

I just reviewed some docs and it sounds like the `is*OfClass` check is a pointer comparison while `respondsToSelector:` can involve a series of checks using the runtime. I've not profiled and the performance difference is probably not worth worrying about for the readability benefit. But, I wanted to double check to see what you think before I make the change.

Reply:
That's interesting. I was thinking more about flexibility/testability/future-proofing/elegance and less about runtime performance. Also, it's nice to be able to pass a "Liskov-substitutable" `NSDictionary` under test without polluting the test environment's `NSUserDefaults`

I'm OK either way -- I'll leave it up to you since ultimately it's going to live in your (separate) library